packages/block-library/src/navigation-link/update-attributes.js

```diff
@@ -87,12 +87,19 @@ export const updateAttributes = (
 		( ! newKind && ! isBuiltInType ) || newKind === 'custom';
 	const kind = isCustomLink ? 'custom' : newKind;
 
+	const hasId = id && Number.isInteger( id );
+
+	// If the new link value has an ID then it's an internal link so do not set the URL.
+	// This will be dynamically generated from the ID.
+	// Passed `url` may already be encoded. To prevent double encoding, decodeURI is executed to revert to the original string.
+	const updatedUrl =
```

Comment:
This causes the LinkControl to display the `Link to empty` message when you first pick a link and the post hasn't yet resolved.